### DISCUSSION TITLE
MQTT5 Features

### DISCUSSION THREAD

**Discussion by RicoBiedermann** [cesanta/mongoose#1964] [Ask questions, get answers] [answered]
Mongoose 7.8 adds MQTT5 support. 
Is it possible to use MQTT5 Features or is it planed to add them? 

I think about: 
- Message and session expiry 
- content type 
- topic alias 
- payload format indicator
- User Properties

**Comment by scaprile** [comment]
Hi, current API is simple: https://mongoose.ws/documentation/#struct-mg_mqtt_opts
and so far only the format differences are supported https://github.com/cesanta/mongoose/blob/master/src/mqtt.c

**Reply by RicoBiedermann** [reply]
Do you know if there is a roadmap or something like that where I can see when the features will be added?

**Comment by scaprile** [comment, marked as answer]
We are currently focused on expanding our supported embedded platforms and rendering our TCP/IP stack (MIP) production quality. Nevertheless, I'll ask. Can you please describe your use case ?

**Reply by RicoBiedermann** [reply]
I want to implement a specification that uses a large number of the features specified with MQTT5. These include those already listed. 
Describing what each feature is actually used for would be a bit complex. 

For example, we need the user properties to append meta-data in case of debug.
As soon as I know the extended authentication mechanism in MQTT5 is more secure than the simple one of MQTT3.1.1.

**Reply by scaprile** [reply]
A use case is a brief description of what your product does and why it does need those MQTT5 features and can't do with currrent MQTT3.11
If you think that is sensitive information, or you are a paying customer and want more support, please write to support@cesanta.com
